linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/FramingFilter.scala

```diff
@@ -0,0 +1,91 @@
+package io.buoyant.linkerd.protocol.http
+
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.linkerd.Headers
+import com.twitter.finagle.http.{Status, _}
+import com.twitter.util.Future
+import com.twitter.logging.Logger
+import io.buoyant.linkerd.ProtocolException
+
+object FramingFilter {
+  // since the client and server filters go in separate stacks,
+  // they can have the same role.
+  val role = Stack.Role("FramingFilter")
+
+  // this is factored out so that the same logic can
+  // be applied to requests from clients and responses from services.
+  private[FramingFilter] def filterMessage[M <: Message](message: M): Future[M] = {
+    val headers = message.headerMap
+    val contentLengths = message.headerMap.getAll("Content-Length")
+
+    if (contentLengths.toSet.size > 1) {
+      // if the length of the Content-Length key in the request/response's
+      // header map is greater than 1, then there are duplicate values.
+      Future.exception(FramingException("conflicting `Content-Length` headers"))
```

Comment:
This exception is always the same so we can store it in a val to avoid allocating a new one each time.